Source/Diagnostics/BoundaryScrapingDiagnostics.cpp

```diff
@@ -0,0 +1,149 @@
+/* Copyright 2022 Remi Lehe
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "BoundaryScrapingDiagnostics.H"
+#include "ComputeDiagFunctors/ComputeDiagFunctor.H"
+#include "Diagnostics/Diagnostics.H"
+#include "Diagnostics/FlushFormats/FlushFormat.H"
+#include "Particles/ParticleBoundaryBuffer.H"
+#include "WarpX.H"
+
+#include <AMReX.H>
+
+#include <string>
+
+using namespace amrex::literals;
+
+BoundaryScrapingDiagnostics::BoundaryScrapingDiagnostics (int i, std::string name)
+    : Diagnostics(i, name)
+{
+    ReadParameters();
+}
+
+void
+BoundaryScrapingDiagnostics::ReadParameters ()
+{
+    BaseReadParameters();
+
+    // Modify some of the quantities that were initialized by default
+    // in the function `BaseReadParameters`
+    m_varnames_fields = {}; // No fields in boundary scraping diagnostics
+    m_varnames = {}; // No fields in boundary scraping diagnostics
+
+    // Number of buffers = 1 for BoundaryScrapingDiagnostics.
+    // (buffers are used in BTDiagnostics, and correspond to different snapshots)
+    m_num_buffers = 1;
+
+    // Do a few checks
+#ifndef AMREX_USE_EB
+    amrex::Abort("You need to compile WarpX with EB, in order to use BoundaryScrapingDiagnostic.");
+#endif
+
+    // Check that saving at EB has been activated for each requested species
+    bool particle_saving_activated = true;
+    for (auto const& species_name : m_output_species_names){
+        amrex::ParmParse pp(species_name);
+        bool save_particles_at_eb;
+        pp.query("save_particles_at_eb", save_particles_at_eb);
+        if (save_particles_at_eb == false) particle_saving_activated = false;
+    }
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE( particle_saving_activated,
+    "You need to set `save_particles_at_eb=1` for each species, in order to use BoundaryScrapingDiagnostic.");
```

Comment:
Let's make this error message more helpful:
- create little list (or [set](https://en.cppreference.com/w/cpp/container/set)) of species names that do not have `save_particles_at_eb` set
- output them of the list is not empty in the user-facing error message.